nmigen_boards/nexys4ddr.py

```diff
@@ -0,0 +1,171 @@
+import os
+import subprocess
+
+from nmigen.build import *
+from nmigen.vendor.xilinx_7series import *
+from .resources import *
+
+
+__all__ = ["Nexys4DDRPlatform"]
+
+
+class Nexys4DDRPlatform(Xilinx7SeriesPlatform):
+    device      = "xc7a100t"
+    package     = "csg324"
+    speed       = "1"
+    default_clk = "clk100"
+    resources   = [
+        Resource("clk100", 0,
+            Pins("E3", dir="i"), Clock(100e6), Attrs(IOSTANDARD="LVCMOS33")),
+
+        *SwitchResources(
+            pins="J15 L16 M13 R15 R17 T18 U18 R13 T8  U8  R16 T13 H6  U12 U11 V10",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+
+        *LEDResources(
+            pins="H17 K15 J13 N14 R18 V17 U17 U16 V16 T15 U14 T16 V15 V14 V12 V11",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+
+        RGBLEDResource(0,
+            r="N15", g="M16", b="R12",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+        RGBLEDResource(1,
+            r="N16", g="R11", b="G14",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+
+        Display7SegResource(0,
+            a="T10", b="R10", c="K16", d="K13", e="P15",
+            f="T11", g="L18", dp="H15", invert=True,
+            attrs=Attrs(iIOSTANDARD="LVCMOS33")),
+        Resource("display_7seg_an", 0,
```

Comment:
What does `_an` mean?

Reply:
`an` stands for ANode. There are 8x7segments multiplexed on this board. At first, I whish I could have define these pins in `Display7SegResource`.

Reply:
Gotcha. I deferred updating Display7SegResource because it was not clear which multiplexing schemes are in use. So this is OK for now.